packages/edit-site/src/components/block-editor/back-button.js

```diff
@@ -22,25 +24,26 @@ function BackButton() {
 	const history = useHistory();
 	const isTemplatePart = location.params.postType === TEMPLATE_PART_POST_TYPE;
 	const isNavigationMenu = location.params.postType === NAVIGATION_POST_TYPE;
-	const previousTemplateId = location.state?.fromTemplateId;
+	// Only show the back button when editing a template part or navigation menu.
+	const canShowBackButton = isTemplatePart || isNavigationMenu;
+	const { setCanvasMode } = unlock( useDispatch( editSiteStore ) );
+	const onClickHandler = () => {
+		if ( location.state?.fromTemplateId ) {
+			history.back();
+		} else {
+			setCanvasMode( 'view' );
+		}
+	};
 
-	const isFocusMode = isTemplatePart || isNavigationMenu;
-
-	if ( ! isFocusMode || ! previousTemplateId ) {
-		return null;
-	}
-
-	return (
+	return canShowBackButton ? (
 		<Button
 			className="edit-site-visual-editor__back-button"
 			icon={ arrowLeft }
-			onClick={ () => {
-				history.back();
-			} }
+			onClick={ onClickHandler }
 		>
 			{ __( 'Back' ) }
```

Comment:
With no `fromTemplateId` context, this will work similar to the WP logo. So we could switch the copy to `__( 'Open navigation' )`, but it sounds a bit naff.